libnmstate/nm/connection.py

```diff
@@ -321,24 +334,40 @@ def _waitfor_active_connection_callback(
             )
 
     @staticmethod
-    def _add_connection_callback(src_object, result, user_data):
+    def _add_connection2_callback(src_object, result, user_data):
         mainloop = user_data
         try:
-            con = src_object.add_connection_finish(result)
+            profile = src_object.add_connection2_finish(result)[0]
```

Comment:
Is this returning a list? I could not understand from the documentation this point (https://lazka.github.io/pgi-docs/NM-1.0/classes/Client.html#NM.Client.add_connection2_finish).

Reply:
ping

Reply:
It return a list when `ignore_out_result=False`. I don't know what this list for, I only care about the first one.

The `ignore_out_result=False` means don't fall back to old `add_connection()` when NM does not support `add_connection2()`.

Reply:
Thank you for the clarification.
This is extremely weird, it will help future us to have this clarified in NM documentation. Can you open a BZ on this?